taipy/gui/utils/patch.py

```diff
@@ -37,22 +37,32 @@ def _patch_value(value: t.Any, change: t.Optional[dict] = None, remove: t.Option
     elif isinstance(value, list):
         if change:
             for k, v in change.items():
-                if isinstance(k, int) and 0 <= k < len(value):
-                    if isinstance(v, dict):
-                        value[k] = _patch_value(value[k], v)
+                if isinstance(k, int):
+                    insert = k < 0
```

Comment:
I wonder if these two lines would read easier in a one-liner:
   insert, idx = True, -k-1 if k < 0 else False, k

Reply:
don't like it, not readable